docs/python_support/python_support.md

```diff
@@ -0,0 +1,664 @@
+                Name of the node in the graph
+            * input_streams: list of strings
+                List of input stream names defined for the node in graph configuration
+            * output_streams: list of strings
+                List of output stream names defined for the node in graph configuration
+        -----------
+        """
+        print("Running initialize...")
+
+    def execute(self, inputs):
+        """
+        `execute` is called in `Process` method of PythonExecutorCalculator which in turn is called by Mediapipe framework. How Mediapipe calls `Process` method for the node depends on the configuration and the two configurations supported by PythonExecutorCalculator are:
+        
+        * Regular: `execute` runs every time the node receives inputs. Produces one set of outputs per one set of inputs. For unary endpoints it's the only possible configuration.
+
+        * Generative: `execute` runs multiple times for single inputs set. Produces multiple sets of outputs over time per single set of inputs. Works only with streaming endpoints. 
+
+        Implemeting this function is required.
+
+        Parameters:
+        -----------
+        * inputs: list of pyovms.Tensor
+        -----------
+        
+        Returns: list of pyovms.Tensor or generator
+        """
+        ...
+        return outputs
+
+    def finalize(self):
+        """
+        `finalize` is called when model server unloads graph definition. It allows to perform any cleanup actions before the graph is removed. Implementing this function is optional.
+        """
+        print("Running finalize...")
+```
+
+### initialize
+
+`initialize` is called when model server loads graph definition. It allows to initialize and maintain state between subsequent `execute` calls and even graph instances.
+
+For gRPC unary, graphs are recreated per request.
+
+For gRPC streaming, there can be multiple graph instances existing at the same time.
+
+`OvmsPythonModel` object is initialized with this method and then shared between all graph instances.
+
+#### Parameters and return value
+
+`initialize` is called with `kwargs` parameter which is a dictionary. All keys are strings. Available keys and values:
+
+| Key           | Value type | Description |
+| ------------- |:-----------| :-----------| 
+| node_name     | string | Name of the node in the graph |
+| input_names   | list of strings | List of input stream names defined for the node in graph configuration | 
+| outputs_names | list of strings | List of output stream names defined for the node in graph configuration |
+
+`initialize` is not expected to return any value.
+
+#### Error handling
+
+Signaling that something went wrong should be done by throwing an exception.
+When model server catches exception from `initialize` it cleans up all Python resources in the graph (including those belonging to the correctly loaded nodes) and sets the whole graph in invalid state.
+
```

Comment:
I noticed that exception from execute is not printed in logs

Reply:
I noticed the same when  there was  error with importing numpy